Src/IronPython/Runtime/PythonEncoding.cs

```diff
@@ -504,4 +591,288 @@ public SurrogateEscapeDecoderFallbackBuffer(bool isPass1, int encodingCharWidth)
 
         }
     }
+
+    internal class PythonSurrogatePassEncoding : PythonEncoding {
+        private const ushort SurrogateRangeStart = 0xd800;
+        private const ushort SurrogateRangeEnd = 0xdfff;
+        private const byte Utf8LeadByte = 0b_1110_0000;
+        private const byte Utf8LeadBytePayload = 0b_1111;
+        private const byte Utf8ContByte = 0b_10_000000;
+        private const byte Utf8ContBytePayload = 0b_111111;
+
+        public PythonSurrogatePassEncoding(Encoding encoding)
+            : base(encoding, new SurrogatePassEncoderFallback(), new SurrogatePassDecoderFallback()) { }
+
+        public class SurrogatePassEncoderFallback : PythonEncoderFallback {
+            public override int MaxCharCount => 1;
+
+            public override EncoderFallbackBuffer CreateFallbackBuffer()
+                => new SurrogatePassEncoderFallbackBuffer(this.IsPass1, this.Encoding);
+        }
+
+        private class SurrogatePassEncoderFallbackBuffer : PythonEncoderFallbackBuffer {
+            private readonly int _codePage;
+            private readonly bool _isBigEndianEncoding;
+
+            public SurrogatePassEncoderFallbackBuffer(bool isPass1, PythonEncoding encoding)
+                : base(isPass1, encoding) {
+                _codePage = encoding.CodePage;
+                _isBigEndianEncoding = encoding.IsBigEndian;
+            }
+
+            public override byte[] GetFallbackBytes(char charUnknown, int index) {
+                if (charUnknown < SurrogateRangeStart || SurrogateRangeEnd < charUnknown) {
+                    // EncoderFallbackException(string, char, int) is not accessible here
+                    throw PythonOps.UnicodeEncodeError(
+                        $"'surrogatepass' error handler can't encode character '{charUnknown}' at index {index}: value not in range(0x{SurrogateRangeStart:x4}, 0x{SurrogateRangeEnd + 1:x4})",
+                        charUnknown,
+                        index
+                    );
+                }
+
+                byte[] fallbackBytes;
+                if (this.EncodingCharWidth > 1) {
+                    fallbackBytes = BitConverter.GetBytes(charUnknown);
+                    if (fallbackBytes.Length == this.EncodingCharWidth) {
+                        // UTF-16LE or UTF-16BE
+                        if (BitConverter.IsLittleEndian == _isBigEndianEncoding) {
+                            // swap bytes for non-native endianness encoding
+                            //(fallbackBytes[0], fallbackBytes[1]) = (fallbackBytes[1], fallbackBytes[0]);
```

Comment:
I think the usage here doesn't end up adding a reference to the `System.ValueTuple` assembly so in theory we could reference the package when building but not include it in the release. Although I'm not seeing a way to prevent using it in a way that adds a reference so if we do that then we might end up using it by accident?

Reply:
Sorry, I meant to delete this comment. I don't think that dragging a whole package into IronPython just to be able to swap two bytes is justified. A `temp` variable does just fine.

Unless you have bigger plans for `ValueTuple` in other places.